src/main/scala/beam/router/RouteHistory.scala

```diff
@@ -73,4 +107,98 @@ class RouteHistory @Inject()() extends LazyLogging {
       }
     }
   }
+  override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
+    val fileHeader = "timeBin,originTAZId,destTAZId,route"
+    val filePath = event.getServices.getControlerIO.getIterationFilename(
+      event.getServices.getIterationNumber,
+      RouteHistory.outputFileBaseName + ".csv.gz"
+    )
+
+    val flattenedRouteHistory: Iterable[(TimeBin, OriginTazId, DestTazId, String)] = routeHistory.flatMap {
+      case (timeBin: TimeBin, origins: TrieMap[OriginTazId, TrieMap[DestTazId, Routes]]) =>
+        origins.flatMap {
+          case (originTazId: OriginTazId, destinations: TrieMap[DestTazId, Routes]) =>
+            destinations.flatMap {
+              case (destTazId: DestTazId, path: Routes) =>
+                Some(timeBin, originTazId, destTazId, path.mkString(":"))
+            }
+        }
+    }
+    val csvContent = flattenedRouteHistory.view
+      .map { tuple =>
+        s"${tuple._1},${tuple._2},${tuple._3},${tuple._4}"
```

Comment:
It might be more readable if you pattern match as part of the map partial function and extract the four components instead of using the underscore notation.

Reply:
Agree